code/game/machinery/portable_turret.dm

```diff
@@ -531,6 +536,11 @@ var/list/turret_icons
 
 	return TURRET_PRIORITY_TARGET	//if the perp has passed all previous tests, congrats, it is now a "shoot-me!" nominee
 
+/obj/machinery/porta_turret/One_star/assess_living(var/mob/living/L)
+	..()
+	if(L.faction == "onestar")
+		return TURRET_NOT_TARGET
```

Comment:
We do not need to call parent proc if the faction check succeeds. Move `..()` after the check and make it `return ..()` because we need to return parent proc's value.